package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -839,51 +815,38 @@ cdef class FastNS(object):
 
         cdef ns_int i, j, size_search
         cdef ns_int d, m
-        cdef ns_int current_beadid, bid
-        cdef ns_int cellindex, cellindex_probe
+        cdef ns_int cellindex_probe
         cdef ns_int xi, yi, zi
 
         cdef NSResults results
 
         cdef dreal d2
         cdef rvec probe
 
-        cdef real[:, ::1] searchcoords
         cdef real[:, ::1] searchcoords_bbox
         cdef _NSGrid searchgrid
         cdef bint check
 
         cdef dreal cutoff2 = self.cutoff * self.cutoff
-        cdef ns_int npairs = 0
-
-        if (search_coords.ndim != 2 or search_coords.shape[1] != 3):
-            raise ValueError("search_coords must have a shape of (n, 3), got "
-                             "{}.".format(search_coords.shape))
 
         # Generate another grid to search
-        searchcoords = search_coords.astype(np.float32, order='C', copy=False)
         searchcoords_bbox = self.box.fast_put_atoms_in_bbox(searchcoords)
-        searchgrid = _NSGrid(searchcoords_bbox.shape[0], self.grid.cutoff, self.box, self.max_gridsize, force=True)
-        searchgrid.fill_grid(searchcoords_bbox)
```

Comment:
The second grid didn't need to get populated, just sized out for later calculations